### ISSUE TITLE
Steam crashes if "Remember Password" is checked

### ISSUE THREAD

**Issue by DIzFer** [NixOS/nixpkgs#34844]
## Issue description

If you check the "Remember Password" option in Steam, it won't launch again after it closes.
I suspect this to be an issue with the dbus session daemon not being reachable from the chrootenv, which may very well point to a misconfiguration on my part. Since dbus is usually left to the DE, so I tried to run steam from within Plasma with the same results. Running Steam from a Solus installation from both Plasma and Budgie sharing home partition works fine.

The output is the following, in which I've broken out the relevant line.
This line _does not appear_ if I the checkbox is off.

```
Running Steam on nixos 18.03.git.c10933d 64-bit
STEAM_RUNTIME has been set by the user to: /steamrt
Pins potentially out-of-date, rebuilding...
mkdir: cannot create directory ‘/nix/store/aqa2807218ya35jbx8df2gsm687dhkij-steam-fhs/steamrt/pinned_libs_32’: Read-only file system
mkdir: cannot create directory ‘/nix/store/aqa2807218ya35jbx8df2gsm687dhkij-steam-fhs/steamrt/pinned_libs_64’: Read-only file system
Installing breakpad exception handler for appid(steam)/version(1513371133)
Installing breakpad exception handler for appid(steam)/version(1513371133)
Installing breakpad exception handler for appid(steam)/version(1513371133)
Installing breakpad exception handler for appid(steam)/version(1513371133)
Installing breakpad exception handler for appid(steam)/version(1513371133)
Installing breakpad exception handler for appid(steam)/version(1513371133)
Installing breakpad exception handler for appid(steam)/version(1513371133)
Installing breakpad exception handler for appid(steam)/version(1513371133)
Installing breakpad exception handler for appid(steam)/version(1513371133)
Installing breakpad exception handler for appid(steam)/version(1513371133)
Installing breakpad exception handler for appid(steam)/version(1513371133)
Installing breakpad exception handler for appid(steam)/version(1513371133)
Installing breakpad exception handler for appid(steam)/version(1513371133)
Installing breakpad exception handler for appid(steam)/version(1513371133)
Installing breakpad exception handler for appid(steam)/version(1513371133)
Installing breakpad exception handler for appid(steam)/version(1513371133)
[0211/142019.312715:ERROR:web_plugin_impl.cc(38)] Widevine registration is not supported after context initialization
Fatal : VkResult is "ERROR_INITIALIZATION_FAILED" in /home/pgriffais/src/Vulkan/base/vulkanexamplebase.cpp at line 823
vulkandriverquery: /home/pgriffais/src/Vulkan/base/vulkanexamplebase.cpp:823: void VulkanExampleBase::initVulkan(): Assertion `res == VK_SUCCESS' failed.
[0211/142020.313405:ERROR:web_plugin_impl.cc(38)] Widevine registration is not supported after context initialization
[0211/142021.314447:ERROR:web_plugin_impl.cc(38)] Widevine registration is not supported after context initialization
[0211/142022.314828:ERROR:web_plugin_impl.cc(38)] Widevine registration is not supported after context initialization
Setting breakpad minidump AppID = 769
Steam_SetMinidumpSteamID:  Caching Steam ID:  76561197960265728 [API loaded no]
[0211/142023.316218:ERROR:web_plugin_impl.cc(38)] Widevine registration is not supported after context initialization
Installing breakpad exception handler for appid(steam)/version(1513371133)
Installing breakpad exception handler for appid(steam)/version(1513371133)
Installing breakpad exception handler for appid(steam)/version(1513371133)
Generating new string page texture 2: 48x256, total string texture memory is 49.15 KB
Generating new string page texture 3: 384x256, total string texture memory is 442.37 KB
Installing breakpad exception handler for appid(steam)/version(1513371133)
roaming config store loaded successfully - 8829 bytes.
migrating temporary roaming config store
Failed to init SteamVR because it isn't installed
```
```
process 31394: arguments to dbus_connection_ref() were incorrect, assertion "connection->generation == _dbus_current_generation" failed in file dbus-connection.c line 2688.
This is normally a bug in some application using the D-Bus library.
  D-Bus not built with -rdynamic so unable to print a backtrace
```
```
[2018-02-11 14:20:14] Startup - updater built Dec 15 2017 11:47:44
Looks like steam didn't shutdown cleanly, scheduling immediate update check
[2018-02-11 14:20:14] Checking for update on startup
[2018-02-11 14:20:14] Checking for available updates...
[2018-02-11 14:20:14] Download skipped: /client/steam_client_ubuntu12 version 1513371133, installed version 1513371133
[2018-02-11 14:20:14] Nothing to do
[2018-02-11 14:20:14] Verifying installation...
[2018-02-11 14:20:14] Performing checksum verification of executable files
[2018-02-11 14:20:15] Verification complete
crash_20180211142014_1.dmp[31830]: Uploading dump (out-of-process)
/tmp/dumps/crash_20180211142014_1.dmp
/home/athena/.local/share/Steam/steam.sh: line 927: 31394 Aborted                 $STEAM_DEBUGGER "$STEAMROOT/$STEAMEXEPATH" "$@"
Refresh rate: 0
Refresh rate: 0
Refresh rate: 0
Refresh rate: 0
Refresh rate: 0
Refresh rate: 0
Refresh rate: 0
Refresh rate: 0
Refresh rate: 0
crash_20180211142014_1.dmp[31830]: Finished uploading minidump (out-of-process): success = yes
crash_20180211142014_1.dmp[31830]: response: CrashID=bp-ced033d1-5640-4e17-8caf-d1e882180211
crash_20180211142014_1.dmp[31830]: file ''/tmp/dumps/crash_20180211142014_1.dmp'', upload yes: ''CrashID=bp-ced033d1-5640-4e17-8caf-d1e882180211''
```

Relevant source of information (that made me notice dbus in the first place) is https://github.com/ValveSoftware/steam-for-linux/issues/3415

While attempting to solve this by myself, I managed to get the dbus-launch command into the steam chrootenv, as well as `eval $(dbus launch)` to be run before steam. It then complains because /etc/dbus-1/{session,system}.conf are dummy files which differ from the host system's (working) files. I think that because of the following output when running `steam-run dbus-launch` is this:

```
Failed to start message bus: Configuration file needs one or more <listen> elements giving addresses
EOF in dbus-launch reading address from bus daemon
```

Finding a way for these files to be equal to the host's (or at least contain a valid config) will probably fix this. However, my Nix knowledge fails me at this point.

### Steps to reproduce

1. run `steam`
2. Enter your username and password
3. Check "Remember Password"
4. Exit Steam
5. Run `steam` again

## Technical details

 - system: `"x86_64-linux"`
 - host os: `Linux 4.15.2, NixOS, 18.03.git.c10933d (Impala)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.0pre5889_c287d731`
 - channels(athena): `""`
 - channels(root): `"nixos-18.03pre115705.c8e7aab0c8"`
 - nixpkgs: `/home/athena/repos/nixpkgs`

**Comment by DIzFer** [comment]
I forgot to mention/test an important detail...

I'm using the native runtime. If I run steam via nix-shell (which I'm pretty sure gives me steam-runtime instead of NixOS'), Steam runs fine. Which makes me think dbus might not be the issue after all. I will try to inspect steam-run's environment and compare it between native and steam runtimes :/

**Comment by abbradar** [comment]
@lbodor if you have a way to unbreak `nativeOnly` please go ahead! The problem was that Steam was segfaulting with a gdb stacktrace being somewhere inside `libdbus-1.so`.

Running a separate dbus session is undesirable but we can have this excuse to finally make `buildFHSEnv` blacklist instead of whitelisting environment variables -- then `DBUS_SESSION_BUS_ADDRESS` would be propagated, problem solved. I'll see what I can do.

Generally I'd prefer to have `nativeOnly` working if only for `steam-run-native` which is very useful in general. Otherwise there's not that much breakage around vendor Steam Runtime actually (I don't remember any serious breakage for a long time actually) -- I don't think we should give up and move to snap just yet.

**Comment by aanderse** [comment]
@DIzFer is this still an issue in latest nixos and steam? Can this ticket be closed?

**Comment by DIzFer** [comment]
Apologies for being unresponsive...

Ok, so, for the record, to reproduce this, I had to put the following in a nixpkgs overlay:
```nix
steam = super.steam.override {
  nativeOnly = true;
};
```

That triggered a warning that the package is broken, so I also had to `allowBroken = true` in nixpkgs.config.

It proceeded to build fine. After several start-close cycles, steam seems to be starting without apparent issues. Or rather, the same issues as regular steam, that being in this case Borderlands 2 is broken again... but I'll probably save that for a merge request :/

Has there been any change to this package that might have fixed this? I kind of lost track a few months ago. In any case, unless there is some other blocker I'm unaware of, the broken flag seems unwarranted now? There probably is :)

In any case, this particular issue seems fixed. Again, I'm sorry for the delay!
